modules/aws-eks-managed-node-groups/iam.tf

```diff
@@ -33,16 +33,8 @@ resource "aws_iam_role_policy_attachment" "managed_ng_AmazonEC2ContainerRegistry
   role       = aws_iam_role.managed_ng.name
 }
 
-# Cluster Autoscaler
-resource "aws_iam_policy" "cluster_autoscaler" {
-  name        = "${var.eks_cluster_id}-${local.managed_node_group["node_group_name"]}-ca"
-  description = "IAM policy for Cluster Autoscaler"
-  path        = var.path
-  policy      = data.aws_iam_policy_document.cluster_autoscaler.json
-}
-
-resource "aws_iam_role_policy_attachment" "cluster_autoscaler" {
-  policy_arn = aws_iam_policy.cluster_autoscaler.arn
+resource "aws_iam_role_policy_attachment" "managed_ng_AmazonSSMManagedInstanceCore" {
+  policy_arn = "${local.policy_arn_prefix}/AmazonSSMManagedInstanceCore"
```

Comment:
Does the worker node actually need this or are we attaching this purely for PVRE purpose? The [template provided in the docs](https://raw.githubusercontent.com/awslabs/amazon-eks-ami/master/amazon-eks-nodegroup.yaml) does not seem to add this.

Reply:
This policy is needed for SSMAgent to work on Node Groups(Managed and Self-managed). This is a key requirement for Node roles used for Karpenter as well. Basic template doesn't provide that.